File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.java
class Solution {
    public String removeOccurrences(String s, String part) {
        StringBuilder ans = new StringBuilder();
        int n = part.length();

        for (char ch : s.toCharArray()) {
            ans.append(ch);

            // if the ending matches part remove it
            if (ans.length() >= n &&
                ans.substring(ans.length() - n).equals(part)) {
                ans.delete(ans.length() - n, ans.length());
            }
        }

        return ans.toString();
    }
}

File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.cpp
class Solution {
public:
    string removeOccurrences(string s, string part) {
        int n=part.size();
        string ans="";
        for(int i=0;i<s.size();i++){
            ans+=s[i];
            if(ans.size()>=part.size() && ans.substr(ans.size()-n)==part){
               ans.erase(ans.size()-n);
            }
            else{
                
            }

        }
        return ans;

    }
};